src/main/java/cs361/battleships/models/Board.java

```diff
@@ -165,27 +197,40 @@ private boolean shipPresent(int x, char y){
 	/*
 	This function takes proposed ship coords and returns true or false based on whether the proposed location is valid
 	 */
-	private boolean validLocation(int size, int x, char y, boolean vertical) {
+	private boolean validLocation(int size, int x, char y, boolean vertical, boolean submerged) {
 		List<Square> allProposedSquares = new ArrayList<>();
+		//if size = 5, it's a sub, so we need to modify the size and set a flag, we'll use an integer flag to modify conditionals on the fly
+		int isSub = 0;
+		if(size == 5) {
+			isSub = 1;
+			size = 4;
+		}
 		//check max x and y and populate proposed squares depending on orientation
 		if (vertical) {
-			if (x + size - 1 > 10 || (int)y > 74) {
+			if (x + size - 1 > 10 || (int) y > (74 - isSub)) {
 				return false;
 			}
 			for (int i = 0; i < size; i++) {
 				allProposedSquares.add(new Square(x + i, y));
 			}
+			//if this is a sub, we'll manually add the "nub", complete with ugly casting to integer and back to char
+			if(isSub==1) allProposedSquares.add(new Square(x+2,(char)((int)y+1)));
 		} else {
-			if ((int) y + size - 1 > 74 || x > 10) {
+			//expanding the bounds checking for when isSub=1
+			if ((int) y + size - 1 > 74 || x > 10 || (x-isSub) < 1) {
```

Comment:
Good addition, we needed to have the submarine fit without changing the meaning behind our "size" variable. Expanding the bounds would let us place our ship near the edges.